src/main/java/edu/agh/wfiis/designpatterns/patternsUsageExample/DesignPattern.java

```diff
@@ -1,102 +1,117 @@
 package edu.agh.wfiis.designpatterns.patternsUsageExample;
 
-class DesignPattern {
+import org.springframework.stereotype.Component;
 
-    private String id;
-
-    private String name;
+public class DesignPattern {
+    private Details patternDetails;
 
     private String patternCode;
 
-    private String description;
-
-    private String example;
-
-    private boolean creational;
-
-    private boolean structural;
-
-    private boolean behavioral;
-
-    public String getId() {
-        return id;
+    public static DesignPattern create(CreatingDesignPattern creator) {
+        return creator.create();
     }
 
-    public void setId(String id) {
-        this.id = id;
+    public Details getPatternDetails() {
+        return patternDetails;
     }
 
-    public String getName() {
-        return name;
-    }
-
-    public void setName(String name) {
-        this.name = name;
-    }
-
-    public String getDescription() {
-        return description;
-    }
-
-    public void setDescription(String description) {
-        this.description = description;
-    }
-
-    public String getExample() {
-        return example;
-    }
-
-    public void setExample(String example) {
-        this.example = example;
-    }
-
-    public boolean isCreational() {
-        return creational;
-    }
-
-    public void setCreational(boolean creational) {
-        this.creational = creational;
-    }
-
-    public boolean isStructural() {
-        return structural;
-    }
-
-    public void setStructural(boolean structural) {
-        this.structural = structural;
-    }
-
-    public boolean isBehavioral() {
-        return behavioral;
-    }
-
-    public void setBehavioral(boolean behavioral) {
-        this.behavioral = behavioral;
-    }
-
-    public String getPatternCode() {
-        return patternCode;
-    }
+    public String refactor(String badCode){
```

Comment:
This method should be extracted to another class because it uses patternCode only. It has access to patternDetails and it is possible to modify this object that is inappropriate. Cohesion is low.